platform/android/java/lib/src/org/godotengine/godot/GodotIO.java

```diff
@@ -461,6 +465,28 @@ public int getScreenDPI() {
 		return (int)(metrics.density * 160f);
 	}
 
+	public int[] getWindowSafeArea() {
+		DisplayMetrics metrics = activity.getResources().getDisplayMetrics();
+		Display display = activity.getWindowManager().getDefaultDisplay();
+		Point size = new Point();
+		display.getRealSize(size);
+
+		int result[] = { 0, 0, size.x, size.y };
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
```

Comment:
All my calls to >28 API stuff are within this conditional block, but perhaps I have an incorrect understanding of what this check does? :thinking:

Reply:
The problem is that it still won't compile if you build with API level < 28 (unless I'm missing something). So it might require some kind of preprocessor condition, but I have no idea how to do it in java.

Reply:
Looking at https://developer.android.com/training/basics/supporting-devices/platforms#java, it seems like this is actually the correct way to do these runtime checks.

We compile with min API 21 and target API 29 currently, so it means while *compiling*, the API 29 SDK is used, but then at runtime the relevant code will not be called on API < 28 here.

So I guess it's good :+1:

Reply:
I did some brief research, and found this: https://android.jlelse.eu/android-api-level-backward-and-forward-compatibility-10e6d31cb848

At least as I understand it, the `compileSdkVersion` determines which methods/types are available during compilation, but the `minSdkVersion` determines the minimum required API level required for the app to run on a device. As long as you perform adequate runtime checks before calling API methods which may not be available on the device, then everything should be cool.

I just tested this change on my old Nexus 5 (running Marshmallow / API 23) and `OS.get_window_safe_area()` works as expected (returning the full screen rectangle). :man_shrugging:

Reply:
Sounds good! I've learned something today :)

Reply:
Me too, thanks for making me double check :)